components/table/util.ts

```diff
@@ -16,13 +17,35 @@ export function getColumnPos(index: number, pos?: string) {
   return pos ? `${pos}-${index}` : `${index}`;
 }
 
+/**
+ * Get first text content in Element
+ *
+ * @param node
+ * @returns
+ */
+function getElementFirstTextContent(node: React.ReactElement): string {
+  if (!node || !node.props || !node.props.children) return '';
+  if (typeof node.props.children === 'string') return node.props.children;
+  return (
+    node.props.children?.map?.((item: React.ReactElement) =>
+      getElementFirstTextContent(item),
+    )?.[0] || ''
+  );
+}
+
 export function renderColumnTitle<RecordType>(
   title: ColumnTitle<RecordType>,
   props: ColumnTitleProps<RecordType>,
 ) {
   if (typeof title === 'function') {
     return title(props);
   }
+  // fix: #38155
+  if (React.isValidElement(title)) {
+    // if title is a React Element, we should get first text content as result,
+    // if there has not text content in React Element, return origin title
```

Comment:
Do we really need this comments? Code seems to be self-explanatory in this case :)